src/main/java/seedu/address/commons/util/DateUtil.java

```diff
@@ -22,4 +23,29 @@ public static LocalDateTime getFirstDayOfMonth(Month month, Year year) {
         requireAllNonNull(month, year);
         return LocalDateTime.of(year.getYear(), month.getMonth(), 1, 0, 0);
     }
+
+    /**
+     * Parse the string input in the form yyyy-... to get the year
+     * @param dateValue string input in yyyy-... form
+     * @return true if year is in yyyy form and precedes a -
+     * @throws ParseException
+     */
+    public static int parseYear(String dateValue) throws ParseException {
+        try {
+            String[] dateValueSplit = dateValue.split("-", 2);
+            int year = Integer.parseInt(dateValueSplit[0]);
+            return year;
+        } catch (NumberFormatException | NullPointerException e) {
+            throw new ParseException("invalid date input");
+        }
+    }
+
+    /**
+     * Checks if given {@code year} will not give the 2038 problem
+     * @param year integer
+     * @return true if {@code year} is between 1970 and 2037 inclusive.
+     */
+    public static boolean pass2038Problem(int year) {
```

Comment:
Haha I think although we have this check because of the 2038 problem, but then I think its quite weird to have this as the method name ah, maybe just using isValidYear(int year) wil be better. Can I check will this method be used tou, don't really see it being used based on this PR, not sure if I have missed it.

Reply:
yeah its used in line 143 of sessionDate
true that will change the method name

Reply:
Ahhh right! Didn't see it! Thanks